src/ESPeasyGPIO.ino

```diff
@@ -322,14 +327,21 @@ void sendMonitorEvent(const char* prefix, byte port, int8_t state)
   rulesProcessing(eventString);
 }
 
-bool checkValidPortRange(byte pluginID, byte port)
+bool checkValidPortRange(pluginID_t pluginID, byte port)
 {
   bool returnValue = false;
   switch (pluginID)
   {
     case PLUGIN_GPIO:
-      returnValue=(port>=0 && port<=PIN_D_MAX);
+    #if defined(ESP8266)
+      returnValue=((port>=0 && port<=5) || (port>=12 && port<=PIN_D_MAX));
+    #elif defined(ESP32)
+      returnValue=((port>=0 && port<=19) || (port>=21 && port<=23) || (port>=25 && port <= 27) || (port>=32 && port<=PIN_D_MAX));
+    #else
+      returnValue=false;
+    #endif //#if defined(ESP8266)
```

Comment:
Why not call `getGpioInfo` here? There is already all knowledge present for all internal GPIO's

Reply:
something like this?
```
    case PLUGIN_GPIO:
      int  pinnr = -1;
      bool input, output, warning;
      returnValue=getGpioInfo(port, pinnr, input, output, warning)) {
    
    break;

```

It just look to me that getGpioInfo is quite a complex function that has many features that are not needed

Reply:
```c++
    case PLUGIN_GPIO:
    {
      int  pinnr = -1;
      bool input, output, warning;
      returnValue=getGpioInfo(port, pinnr, input, output, warning);
    }    
    break;
```

As you will create new variables within the scope of the case, use { }

And yes it does more than strictly needed, but at least the knowledge of what pin is used where is only defined in a single place in the code.
This is the kind of information you for sure only want to have it occur once in the code so if there is ever a bug in there, you can fix it in one place and not run into different situations where the same bug may apply also but only in some very different use cases.

If the function is too complex, then we can split it into 2 parts like "is it valid" and "anything special about the pin?"
But please have it only defined in 1 spot.
As you may know I may be sometimes a bit too OCD about DRY (Don't Repeat Yourself), but for these kind of things where the very specific knowledge is used in a number of places you really must try to keep it defined in only a single spot in the code or else you will loose countless hours debugging and fixing.

Reply:
done